src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -2069,6 +2071,28 @@ public VolumeList listVolumes(ListVolumesParam... params)
     return request(GET, VolumeList.class, resource, resource.request(APPLICATION_JSON_TYPE));
   }
 
+  @Override
+  public SecretCreateResponse createSecret(final SecretSpec secret) 
+      throws DockerException, InterruptedException {
+    assertApiVersionIsAbove("1.25");
+    final WebTarget resource = resource().path("secrets").path("create");
+
+    try {
+      return request(POST, SecretCreateResponse.class, resource,
+                     resource.request(APPLICATION_JSON_TYPE),
+                     Entity.json(secret));
+    } catch (DockerRequestException dre) {
+      switch (dre.status()) {
+        case 409:
+          throw new DockerException("Name conflicts with an existing object.", dre);
+        case 503:
```

Comment:
Is this correct? According to the [API docs](https://docs.docker.com/engine/api/v1.25/#operation/SecretCreate), this should be 406, but the docs have been known to be wrong.